server.go

```diff
@@ -118,45 +118,29 @@ func (s *SockchatServer) webSocket(w http.ResponseWriter, r *http.Request) {
 			continue
 		}
 		conn.SetReadDeadline(time.Now().Add(s.timeoutAuthorized))
+		var req any
 		switch receivedMsg.Action {
-		case CreateAction:
-			c := UnmarshalChannelRequest(receivedMsg.Payload)
-			if c == nil {
-				conn.WriteSocketMsg(NewErrorMessage(InvalidRequestEvent))
-				continue
-			}
-			if err := handler.RequestNewChannel(*c, s.channelStore); err != nil {
-				conn.WriteSocketMsg(NewErrorMessage(err.Error()))
-				continue
-			}
-			conn.WriteSocketMsg(NewSocketMessage("channel_created", ChannelRequest{c.Name}))
-
-		case JoinAction:
-			c := UnmarshalChannelRequest(receivedMsg.Payload)
-			if c == nil {
-				conn.WriteSocketMsg(NewErrorMessage(InvalidRequestEvent))
-				continue
-			}
-			if err := handler.RequestJoinChannel(*c, s.channelStore); err != nil {
-				conn.WriteSocketMsg(NewErrorMessage(err.Error()))
-			}
-		case LeaveAction:
-			if c := UnmarshalChannelRequest(receivedMsg.Payload); c != nil {
-				err = handler.RequestLeaveChannel(*c, s.channelStore)
-			} else {
-				conn.WriteSocketMsg(NewErrorMessage(InvalidRequestEvent))
-			}
+		case CreateAction, JoinAction, LeaveAction:
+			req, err = UnmarshalChannelRequest(receivedMsg.Payload)
 		case SendMessageAction:
-			if m := UnmarshalMessageRequest(receivedMsg.Payload); m != nil {
-				err = handler.RequestSendMessage(*m, s.channelStore)
-			} else {
-				conn.WriteSocketMsg(NewErrorMessage(InvalidRequestEvent))
-			}
+			req, err = UnmarshalMessageRequest(receivedMsg.Payload)
+		case LoginAction:
+			// Login request is handled in authorizeConnection
+			continue
+		default:
+			conn.WriteSocketMsg(NewErrorMessage(InvalidRequestEvent))
+			continue
 		}
 		if err != nil {
-			conn.WriteSocketMsg(NewErrorMessage(err.Error()))
+			log.Print("error while unmarshaling request: ", err.Error())
+			conn.WriteSocketMsg(NewErrorMessage(InvalidRequestEvent))
+			continue
 		}
-
+		func(err error) {
+			if err != nil {
+				conn.WriteSocketMsg(NewErrorMessage(err.Error()))
+			}
+		}(handler.MakeRequest(receivedMsg.Action, req))
```

Comment:
```suggestion
		err = handler.MakeRequest(receivedMsg.Action, req)
		if err != nil {
			conn.WriteSocketMsg(NewErrorMessage(err.Error()))
		}
```
It can be simplified